src/cljs/athens/views/db_menu/core.cljs

```diff
@@ -0,0 +1,150 @@
+    [reagent.core :as r]
+    [stylefy.core :as stylefy :refer [use-style]]))
+
+
+;; -------------------------------------------------------------------
+;; --- material ui ---
+
+(def m-popover (r/adapt-react-class (.-default Popover)))
+
+
+;; temporary local defs
+
+;; Save last visited db in local storage and get current-db path from local storage on startup
+(def current-db-path "ec2-3-16-89-123.us-east-2.compute.amazonaws.com")
+
+
+;; make this a subscription handler, and subscribe to this in db-menu
+(def dummy-all-dbs
+  [{:name "Athens Test Remote DB"
+    :path "ec2-3-16-89-123.us-east-2.compute.amazonaws.com"
+    :token "x"
+    :is-remote true}
+   {:name "My DB"
+    :path "/Users/coolUser/Documents/athens/index.transit"
+    :is-remote false}
+   {:name "Top Secret"
+    :path "/Users/coolUser/Documents/athens2/index.transit"
+    :is-remote false}])
+
+
+;; Style
+
+(def dropdown-style
+  {::stylefy/manual [[:.menu {:background (color :background-plus-2)
+                              :color (color :body-text-color)
+                              :border-radius "calc(0.25rem + 0.25rem)" ; Button corner radius + container padding makes "concentric" container radius
+                              :padding "0.25rem"
+                              :display "inline-flex"
+                              :box-shadow [[(:64 DEPTH-SHADOWS) ", 0 0 0 1px rgba(0, 0, 0, 0.05)"]]}]]})
+
+
+(def db-menu-button-style
+  {:color (color :body-text-color :opacity-high)
+   :background "inherit"
+   :padding "0"
+   :border "1px solid transparent"
+   ::stylefy/manual [["&:hover" {:filter "brightness(110%)"}]
+                     ["&:active" {:filter "brightness(90%)"}]
+                     [:&.is-active {:color (color :body-text-color);
+                                    :filter "brightness(90%)"}]
+                     [:.icon {:width "1.75em"
+                              :height "1.75em"}]]})
+
+
+(def current-db-area-style
+  {:background "rgba(144, 144, 144, 0.05)"
+   :margin "-0.25rem -0.25rem 0.125rem"
+   :border-bottom [["1px solid " (color :border-color)]]
+   :padding "0.25rem"})
+
+
+(def current-db-tools-style
+  {:margin-left "2rem"})
+
+
+;; Components
+
+(defn current-db-tools
+  ([{:keys [db]} all-dbs]
+   [:div (use-style current-db-tools-style)
+    (if (:is-remote db)
+      [:<>
+       [button "Import"]
+       [button "Copy Link"]
+       [button "Remove"]]
+      [:<>
+       [button {:onClick #(electron/move-dialog!)} "Move"]
+    ;; [button {:onClick "Rename"]
+       [button {:onClick #(if (= 1 (count all-dbs))
+                            (js/alert "Can't remove last db from the list")
+                            (do
+                              (dispatch [:db-picker/remove-db-from-list (:path db)])
+                              (dispatch [:db-picker/delete-db (:path db)])))}
+               "Delete"]])]))
+
+
+(defn db-menu
+  []
+  (r/with-let [ele (r/atom nil)]
```

Comment:
`ele` is used for 2 purposes, which often is surprising when reading code.

1st usage as DOM element itself in `:anchorEl`, to get DOM element of reagent component use `:ref` (correct way of doing this)[https://github.com/reagent-project/reagent/blob/master/doc/FAQ/UsingRefs.md#answer]

2nd usage is predicate if menu is open. it's better to express it explicitly as so, or maybe even manage it re-frame itself?

Reply:
@shanberg